isic_challenge_scoring/load_csv.py

```diff
@@ -27,11 +27,19 @@ def parse_truth_csv(csv_file_stream: TextIO) -> Tuple[pd.DataFrame, pd.DataFrame
 
 
 def parse_csv(csv_file_stream: TextIO, categories: pd.Index) -> pd.DataFrame:
+    if csv_file_stream.read(2000).count('\n') < 2:
+        # Heuristic: if there aren't 2 newlines in the first 2000 characters, it's probably invalid,
+        # and we don't want to hang or crash the parser
+        raise ScoreException('No newlines detected in CSV.')
+    csv_file_stream.seek(0)
+
     try:
         probabilities = pd.read_csv(csv_file_stream, header=0, index_col=False)
-    except pd.errors.ParserError as e:
-        # TODO: Test this case
-        raise ScoreException(f'Could not parse CSV: "{str(e)}"')
+    except (pd.errors.ParserError, pd.errors.EmptyDataError) as e:
+        # TODO: Test something that generates a ParserError
+        raise ScoreException(f'Could not parse CSV: "{str(e)}".')
+    except IndexError:
+        raise ScoreException('Could not parse CSV: mismatched header row.')
```

Comment:
Can this be a bit more human readable?

Reply:
Do you have a suggestion?

Reply:
"Could not parse CSV: differing number of columns" ?

Reply:
Better now?